sdk/all/src/main/java/io/opentelemetry/sdk/OpenTelemetrySdk.java

```diff
@@ -144,130 +111,28 @@ public Builder setPropagators(ContextPropagators propagators) {
       return this;
     }
 
-    /**
-     * Sets the {@link Clock} to be used for measuring timings.
-     *
-     * <p>If you use {@link #setTracerProvider(TracerProvider)}, this will be ignored for purposes
-     * of configuring the TracerProvider.
-     *
-     * @param clock The clock to use for all temporal needs.
-     * @return this
-     */
-    public Builder setClock(Clock clock) {
```

Comment:
While there could be an argument for removing the tracerprovider knobs here, I don't see why we would remove global concerns like clock / resource.

Reply:
Because they are properties of the providers why would we need to have getters for Clock and Resource in the SDK?

Reply:
Also it is a very poor design choice, if user sets TracerProvider and Resource the Resource is not set to the provider so it is an unexpected behavior for the user

Reply:
It's poor design to make it easy to set a different resource for tracer and meter right? The getters can probably be removed, they were with the thought that the providers might read them from the OpenTelemetry, but they can be pushed down via the builders instead. The idea is to remove `setTracerProvider` / `setMeterProvider` from here to make sure cross-signal concerns are handled by the SDK itself. We're making steps towards that and after making sure SDK can be configured as one unit, and each signal can also be configured as one unit, we can remove them.

Reply:
Not sure about this, I can see reasons to have different clock instances (for metrics you don't need `us` precision, `ms` is more than enough). So the only part that is really common is `Resource`, does it worth all this duplicated code? Also as a user right now I am totally confused, where do I set `Resource`? Where do I set `SpanProcessors`?

1. So far we clearly agreed that the getters even if available to pass to the providers, they do not need to be public.

Do we agree that having duplicate way to configure the same thing is a maintainers problem?

Reply:
Yeah - two ways to do the same thing is an issue of maintenance, which takes the back seat to user experience.

> Also as a user right now I am totally confused, where do I set Resource? Where do I set SpanProcessors?

I don't understand this - the configuration is on the SDK builder, why are you confused? (after we remove setTracerProvider / setMeterProvider)

Reply:
> Yeah - two ways to do the same thing is an issue of maintenance, which takes the back seat to user experience.

But, we already have tons of ways to configure things...env vars, system properties, programmatic, SPI. How do we draw the line at what the "right" number of ways is to configure the API and SDK?